### PR TITLE
feat(component-change-detection): add change detection strategy rule

### LINKED ISSUE #135: Feature request: Enforce value of @Component.changeDetection
In the project I'm currently working on, it'd be useful to ensure that all components are set to use `changeDetection: ChangeDetectionStrategy.OnPush`.

### PR DESCRIPTION
**Pull request by mattlewis92**
Please let me know if there's any changes I need to make! 😄 

Closes #135

### CODE REVIEW

File: src/componentChangeDetectionRule.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+const OPTION_DEFAULT = 'Default';
+const OPTION_ON_PUSH = 'OnPush';
+
+export type ChangeDetectionType = 'Default' | 'OnPush';
+
+export class Rule extends Lint.Rules.AbstractRule {
+  public static metadata: Lint.IRuleMetadata = {
+    ruleName: 'component-change-detection',
+    type: 'functionality',
+    description: 'Enforce the preferred component change detection type (Default or OnPush).',
```
Comment (wKoza) [inline src/componentChangeDetectionRule.ts]: I'm not sure it's necessary to have an option.  We want to have a check on use of `OnPush`, not of `Default`

Comment (mattlewis92) [reply src/componentChangeDetectionRule.ts]: Thanks so much for the quick review! I was debating about having the option as well, I suppose if angular ever offered a 3rd change detection strategy then we could just add an option for that that default it to OnPush so it wasn't breaking. WDYT?

Comment (wKoza) [reply src/componentChangeDetectionRule.ts]: In your implementation, if we use the `DEFAULT` option, we have to set the property `changeDetection: ChangeDetectionStrategy.Default`. It's weird for a default property, no ?

Comment (mattlewis92) [reply src/componentChangeDetectionRule.ts]: Yeah agreed, that is kind of weird, I'll just remove the option and keep it as `OnOpen` 👍

**Comment by mattlewis92** [comment]
@wKoza removed the option, LMK what you think! 😄

**Comment by wKoza** [comment]
LGTM ! Thks!

**Comment by mgechev** [comment]
Let's release this once we introduce support for 8.0.0.

**Pull request merged**